src/store/relationship.js

```diff
@@ -1,25 +1,36 @@
 import Ember from 'ember';
 import Relationship from 'ember-graph/relationship/relationship';
+import RelationshipHash from 'ember-graph/relationship/relationship_hash';
 
 const { CLIENT_STATE, SERVER_STATE, DELETED_STATE } = Relationship;
 
+// ### TODO: 1. incorporate both type AND id into hash (reduce repetitive verifications)
+// ### TODO: 2. move all remaining type== id== tests to relationship/relationships.js
+
 export default {
 
 	allRelationships: Ember.Object.create(),
 
-	queuedRelationships: Ember.Object.create(),
+	relationshipsById: RelationshipHash.create(),
+
+	queuedRelationships: RelationshipHash.create(),
+
+	// oldQueuedRelationships: Ember.Object.create(),
```

Comment:
Agree with losing all the commented out stuff.  This was debugging code (trying to compare the "old" vs. "new" operations to figure out what was getting lost).  Didn't get a chance for a final clean-up on the last optimization set as I was racing to get them working at the end.